docs/user/templates.rst

```diff
@@ -0,0 +1,172 @@
+.. _templates:
+
+Templates in Renku
+==================
+
+Templates come in handy when you want to be able to use the same kind
+of high level project structure, dependencies, and/or scripts layouts over and
+over. The two main aspects of the project that can be templated are:
+
+* the directory structure & files
+* the Docker image that builds your environment
+
+When you create a project on the Renku platform, you can choose between two base
+templates: a minimal python setup, and a minimal R setup. In many cases, you can
+use these templates as-is (see the :ref:`directory_structure` below). If the base
+template for either of these languages is not sufficient for you, or you require
+a different IDE, you can make modifications! Check out :ref:`advanced_interfaces`.
+
+Note that you can create projects that are specifically intended to be templates
+for other projects.
+
+.. contents:: :local:
+
+.. _directory_structure:
+
+Directory Structure
+^^^^^^^^^^^^^^^^^^^
+
+The default provided directory structure is approximately the following:
+TODO: TURN THIS INTO A SET OF LINKS TO THE BELOW SECTION
+
+ .renku
+ data
+ -- .gitkeep
+ notebooks
+ -- .gitkeep
+ .dockerignore
+ .gitignore
+ .gitlab-ci.yml
+ Dockerfile
+ README.md
+ environment.yml
+ requirements.txt
+
+data and notebooks
+""""""""""""""""""
+
+The ``data`` and ``notebooks`` directories are where you should keep your...
+data (added via ``renku dataset``) and notebooks (by double-clicking on a python3
+or R kernel image from the jupyterlab instance while you're inside the dir), respectively.
+You can add further nesting of directories under these locations to keep your
+project organized.
+
+Moreover, you will probably want to create other top-level directories, like ``src`` for
+keeping scripts that you create from your notebooks when your analysis stabilizes,
+and ``docs`` if you wish to keep your documentation separate from the analysis.
+See :ref:`how_to_documentation` for documentation tips.
+
+The ``.gitkeep`` files in these directories by default are a convention used to
+git commit "empty" directories (where normal git behavior is to omit empty
+directories).
+
+environment.yml and requirements.txt
+""""""""""""""""""""""""""""""""""""
+
+These two files are where you write in your conda (environment.yml) and pip
+(requirements.txt) library dependencies. This is so that when you kill a notebook
+or anyone forks your project, (re)starting an interactive environment session
+will pre-install your libraries.
+
+To make it easy to remember to write your dependencies to this file, instead of
+``pip install <library>`` in a terminal, for instance, you can get into the habit
+of running ``pip install -r requirements.txt``.
+
```

Comment:
Maybe we want to mention the option of using `pip freeze > requirements.txt` to lock down the dependencies? Though it can lead to irrecoverable problems.

Reply:
Yeah -- and also just ran into a bunch of R environment issues so we probably want to make recommendations for this as well (once we figure out what they are)

Reply:
yep, this can be an iterative process :)